sdk/python/kfp/local/e2e_test.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2023 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""E2E local execution tests.
+
+These contain tests of various component definitions/types, tested on each local runner type + configurations.
+
+These can be thought of as local runner conformance tests. The test results should be the same irrespective of the runner.
+"""
+import io
+import sys
+from typing import NamedTuple
+import unittest
+
+from absl.testing import parameterized
+from kfp import dsl
+from kfp import local
+from kfp.dsl import Artifact
+from kfp.dsl import Dataset
+from kfp.dsl import Output
+from kfp.local import testing_utilities
+
+# NOTE when asserting on task.output or task.outputs[<key>]
+# since == is overloaded for dsl.Condition, if local execution is not
+# "hit", then actual will be a channel and actual == expected evaluates
+# to ConditionOperation. Since ConditionOperation is truthy,
+# this may result in a false negative test result. For this reason,
+# we perform an isinstance check first.
+
+ALL_RUNNERS = [
+    (local.SubprocessRunner(use_venv=False),),
+    (local.SubprocessRunner(use_venv=True),),
+]
+
+
+@parameterized.parameters(ALL_RUNNERS)
+class TestLightweightPythonComponentLogic(
+        testing_utilities.LocalRunnerEnvironmentTestCase):
+
+    def test_str_input(self, runner):
+        local.init(runner=runner)
+
+        @dsl.component
+        def identity(x: str) -> str:
+            return x
+
+        actual = identity(x='hello').output
+        expected = 'hello'
+        self.assertIsInstance(actual, str)
```

Comment:
nit: seems a bit redundant to me as the equality test next should cover this as well. 
But I don't have strong opinion, feel free to keep it.

Reply:
I added [a comment](https://github.com/kubeflow/pipelines/blob/4e3699e4dcbbe3a84204213c79c71d9c66c4d418/sdk/python/kfp/local/e2e_test.py#L33-L38) to the top of the file on why this is needed. Added to the top, since it applies for every assertion.